### PR TITLE
Pods 2964

### PR DESCRIPTION
**Pull request by suryabalasingh**

### CODE REVIEW

File: test/controllers/register/trustees/individual/CheckYourAnswersIndividualContactDetailsControllerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+        StringCYA[TrusteePhoneId](
+         Some(messages("messages__common_phone__heading", trusteeName)),
+          Some(messages("messages__common_phone__visually_hidden_change_label", trusteeName))
+        )().row(TrusteePhoneId(index))(
+          routes.TrusteePhoneController.onPageLoad(checkMode(mode), Index(index), srn).url, userAnswers)
+    ))
+  }
+
+  def controller(dataRetrievalAction: DataRetrievalAction = getEmptyData,
```
Comment (amaalali) [inline test/controllers/register/trustees/individual/CheckYourAnswersIndividualContactDetailsControllerSpec.scala]: We should probably be using `applicationBuilder` here to make the intent of the code clearer and get rid of the mechanics of creating instances.
https://github.com/hmrc/pensions-scheme-frontend/blob/master/test/base/SpecBase.scala#L60-L69

File: app/controllers/register/trustees/individual/CheckYourAnswersIndividualContactDetailsController.scala [hunk 2]
```diff
@@ -19,34 +19,59 @@ package controllers.register.trustees.individual
...
+                                                                   implicit val countryOptions: CountryOptions,
+                                                                   allowChangeHelper: AllowChangeHelper
                                      )(implicit val ec: ExecutionContext) extends FrontendController with Retrievals with I18nSupport with Enumerable.Implicits {
 
   def onPageLoad(mode: Mode, index: Index, srn: Option[String]): Action[AnyContent] =
-    (authenticate andThen getData(mode, srn) andThen allowAccess(srn) andThen requireData) {
-      implicit request => NotImplemented("Not implemented: " + this.getClass.toString)
+    (authenticate andThen getData(mode, srn) andThen allowAccess(srn) andThen requireData).async {
+      implicit request =>
+        implicit val userAnswers: UserAnswers = request.userAnswers
```
Comment (amaalali) [inline app/controllers/register/trustees/individual/CheckYourAnswersIndividualContactDetailsController.scala]: We don't seem to need to have this as an implicit value. Also, it only seems to be used on the line below so I'm not sure it's necessary

File: test/controllers/register/trustees/individual/CheckYourAnswersIndividualContactDetailsControllerSpec.scala [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+class CheckYourAnswersIndividualContactDetailsControllerSpec extends ControllerSpecBase with ControllerAllowChangeBehaviour {
+
+  private val index = Index(0)
+  private val srn = Some("test-srn")
+  private implicit val fakeCountryOptions: CountryOptions = new FakeCountryOptions
+
+  private def submitUrl(mode: Mode = NormalMode, srn: Option[String] = None): Call =
+    routes.CheckYourAnswersIndividualContactDetailsController.onSubmit(mode, index, srn)
+
+  private def answerSection(mode: Mode, srn: Option[String] = None)(implicit request: DataRequest[AnyContent]): Seq[AnswerSection] = {
```
Comment (amaalali) [inline test/controllers/register/trustees/individual/CheckYourAnswersIndividualContactDetailsControllerSpec.scala]: I don't think we should make `request` an implicit parameter here, it's convenient but is treading a bit close to being _mystery guest_-y when used on lines 99 and 107

File: app/identifiers/register/trustees/company/CompanyEmailId.scala [hunk 4]
```diff
@@ -38,7 +38,7 @@ object CompanyEmailId {
     override def row(id: CompanyEmailId)(changeUrl: String, userAnswers: UserAnswers): Seq[AnswerRow] = {
       def trusteeName(index: Int) = userAnswers.get(CompanyDetailsId(index)).fold(messages("messages__theTrustee"))(_.companyName)
       def label(index:Int) = messages("messages__common_email__heading", trusteeName(index))
-      def hiddenLabel(index:Int) = Some(messages("messages__common_trustee_company_email__visually_hidden_change_label", trusteeName(index)))
+      def hiddenLabel(index:Int) = Some(messages("messages__common_email__visually_hidden_change_label", trusteeName(index)))
```
Comment (amaalali) [inline app/identifiers/register/trustees/company/CompanyEmailId.scala]: These don't seem to need to be functions (Int => String). We can just have on line 39:
```
val trusteeName = userAnswers.get(CompanyDetailsId(id.index)).fold(messages("messages__theTrustee"))(_.companyName)
```
and then here becomes
```
val hiddenLabel = Some(messages("messages__common_email__visually_hidden_change_label", trusteeName))```

File: test/identifiers/register/trustees/individual/TrusteeEmailIdSpec.scala [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    implicit val countryOptions: CountryOptions = new CountryOptions(Seq.empty[InputOption])
+    val email = "test@test.com"
+    val onwardUrl = "onwardUrl"
+    val personName = PersonName("first", "last")
+
+    Seq(NormalMode, UpdateMode).foreach { mode =>
+
+      s"in ${mode.toString} mode" must {
+        "return answers rows with change links" in {
+          implicit val request: DataRequest[AnyContent] = DataRequest(FakeRequest(), "id",
```
Comment (amaalali) [inline test/identifiers/register/trustees/individual/TrusteeEmailIdSpec.scala]: I think its clearer if we explicitly pass these as params

File: app/controllers/register/trustees/individual/CheckYourAnswersIndividualContactDetailsController.scala (hunk 2 above)
Comment (suryabalasingh) [reply app/controllers/register/trustees/individual/CheckYourAnswersIndividualContactDetailsController.scala]: Its being used by the id below

Comment (amaalali) [reply app/controllers/register/trustees/individual/CheckYourAnswersIndividualContactDetailsController.scala]: Right makes sense, didn't spot that before 👍

**Pull request merged**